rclcpp/src/rclcpp/node_interfaces/node_topics.cpp

```diff
@@ -47,6 +47,14 @@ NodeTopics::create_publisher(
     // Get the intra process manager instance for this context.
     auto ipm = context->get_sub_context<rclcpp::intra_process_manager::IntraProcessManager>();
     // Register the publisher with the intra process manager.
+    if (publisher_options.qos.history == RMW_QOS_POLICY_HISTORY_KEEP_ALL) {
+      throw exceptions::InvalidParametersException(
```

Comment:
I don't think this is the correct error type. This error is meant to be used specifically with "ROS Parameters" not any old C++ argument/parameter. I think `std::invalid_argument` (https://en.cppreference.com/w/cpp/error/invalid_argument) is more appropriate.

Reply:
You're right. Corrected it in `setup_intra_process` method too.
Also, added tests for checking if it's actually throwing an exception.